core/src/test/java/org/apache/iceberg/TestSnapshotJson.java

```diff
@@ -105,6 +129,40 @@ public void testJsonConversionWithOperation() throws IOException {
     assertThat(snapshot.operation()).isEqualTo(expected.operation());
     assertThat(snapshot.summary()).isEqualTo(expected.summary());
     assertThat(snapshot.schemaId()).isEqualTo(expected.schemaId());
+    assertThat(snapshot.firstRowId()).isNull();
+    assertThat(snapshot.addedRows()).isNull();
+  }
+
+  @Test
+  public void testJsonConversionWithRowLineage() throws IOException {
+    int snapshotId = 23;
+    Long parentId = null;
+    Long firstRowId = 20L;
+    Long addedRows = 30L;
+    String manifestList = createManifestListWithManifestFiles(snapshotId, parentId);
+
+    Snapshot expected =
+        new BaseSnapshot(
+            0,
+            snapshotId,
+            parentId,
+            System.currentTimeMillis(),
+            null,
+            null,
+            null,
+            manifestList,
+            firstRowId,
+            addedRows);
+    String json = SnapshotParser.toJson(expected);
+    Snapshot snapshot = SnapshotParser.fromJson(json);
+
+    assertThat(snapshot.snapshotId()).isEqualTo(expected.snapshotId());
+    assertThat(snapshot.allManifests(ops.io())).isEqualTo(expected.allManifests(ops.io()));
+    assertThat(snapshot.operation()).isNull();
+    assertThat(snapshot.summary()).isNull();
+    assertThat(snapshot.schemaId()).isNull();
+    assertThat(snapshot.firstRowId()).isEqualTo(firstRowId);
+    assertThat(snapshot.addedRows()).isEqualTo(addedRows);
   }
```

Comment:
Do we have tests for snapshots produced where tables where there is no row lineage enabled?

Reply:
Ah missed the assertions above this test for the previous test, great!